playwright/_impl/_network.py

```diff
@@ -247,37 +294,86 @@ async def fulfill(
+        self,
+        url: str = None,
+        method: str = None,
+        headers: Dict[str, str] = None,
+        postData: Union[str, bytes] = None,
+    ) -> None:
+        overrides = cast(FallbackOverrideParameters, locals_to_params(locals()))
+        self._check_not_handled()
+        self.request._apply_fallback_overrides(overrides)
+        await self._internal_continue()
+        self._report_handled(True)
+
+    def _internal_continue(
+        self, is_internal: bool = False
+    ) -> Coroutine[Any, Any, None]:
         async def continue_route() -> None:
             try:
-                await self.continue_()
-            except Exception:
-                pass
-
-        asyncio.create_task(continue_route())
+                post_data_for_wire: Optional[str] = None
+                post_data_from_overrides = self.request._fallback_overrides.get(
+                    "postData"
+                )
+                if post_data_from_overrides is not None:
+                    post_data_for_wire = (
+                        base64.b64encode(post_data_from_overrides.encode()).decode()
+                        if isinstance(post_data_from_overrides, str)
+                        else base64.b64encode(post_data_from_overrides).decode()
+                    )
+                await self._race_with_page_close(
+                    self._channel.send(
+                        "continue",
+                        cast(
+                            Any,
+                            locals_to_params(
+                                {
+                                    "url": self.request._fallback_overrides.get("url"),
+                                    "method": self.request._fallback_overrides.get(
+                                        "method"
+                                    ),
+                                    "headers": serialize_headers(
+                                        cast(
+                                            Dict[str, str],
+                                            self.request._fallback_overrides["headers"],
+                                        )
+                                    )
+                                    if self.request._fallback_overrides.get("headers")
+                                    else None,
+                                    "postData": post_data_for_wire,
+                                }
+                            ),
+                        ),
+                    )
+                )
+            except Exception as e:
+                if not is_internal:
+                    raise e
+
+        return continue_route()
+
+    # FIXME: Port corresponding tests, and call this method
+    async def _redirected_navigation_request(self, url: str) -> None:
+        self._check_not_handled()
+        await self._race_with_page_close(
+            self._channel.send("redirectNavigationRequest", {"url": url})
+        )
+        self._report_handled(True)
```

Comment:
This FIXME will be removed when https://github.com/microsoft/playwright-python/pull/1384 lands (since that is the PR that actually uses this particular method).